pcmanfm/desktopwindow.cpp

```diff
@@ -810,8 +811,13 @@ bool DesktopWindow::pickWallpaper() {
                        files.removeOne(ls); // choose from other images
                }
                // this is needed for the randomness, especially when choosing the first wallpaper
-               qsrand((uint)QTime::currentTime().msec());
-               int randomValue = qrand() % files.size();
+
+               // Conversion notes:
+               // The random generated value can be safely converted to an int because it's max
+               // value is also an int (files.size() - 1).
+               // files.size() - 1 is always => 1
+               // files.at(0) -> valid, files.at(size) -> invalid
+               int randomValue = static_cast<int>(QRandomGenerator::global()->bounded(static_cast<quint32>(files.size() - 1)));
```

Comment:
This approach is very good. Just one thing: `QRandomGenerator::bounded(quint32 highest)` excludes `highest`. So, its argument needs to be `files.size()` in our case; otherwise, the last image will never be seen.

Reply:
read 256 instead of 255

Reply:
Yes, if we want 0-255, we set 256. Here, we want a number in the range 0-[files.size() - 1]; so, we have to use `files.size()`.